From e8200b428df61f8c04a8bb832c0f756d81d33779 Mon Sep 17 00:00:00 2001
From: Suresh Gupta <suresh.gupta@xilinx.com>
Date: Fri, 5 Oct 2018 19:42:51 +0530
Subject: [PATCH 0444/1852] v4l: xilinx-multi-scaler: Remove unsupported
 formats

commit 0a8940c203df8943fbc3bd74b7969d47ed5dfd73 from
https://github.com/Xilinx/linux-xlnx.git

The Multi-Scaler IP do not support Alpha formats and
planes which are non-contiguous in memory, so removing
those formats.

Signed-off-by: Suresh Gupta <suresh.gupta@xilinx.com>
Reviewed-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../platform/xilinx/xilinx-multi-scaler.c     | 80 ++++---------------
 1 file changed, 16 insertions(+), 64 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-multi-scaler.c b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
index 26daecc13190..d462065a7362 100644
--- a/drivers/media/platform/xilinx/xilinx-multi-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
@@ -99,25 +99,22 @@
 
 /* Xilinx Video Specific Color/Pixel Formats */
 enum xm2msc_pix_fmt {
-	XILINX_M2MSC_FMT_RGBX8 = 10,
-	XILINX_M2MSC_FMT_YUVX8,
-	XILINX_M2MSC_FMT_YUYV8,
-	XILINX_M2MSC_FMT_RGBA8,
-	XILINX_M2MSC_FMT_YUVA8,
-	XILINX_M2MSC_FMT_RGBX10,
-	XILINX_M2MSC_FMT_YUVX10,
-	XILINX_M2MSC_FMT_Y_UV8 = 18,
-	XILINX_M2MSC_FMT_Y_UV8_420,
-	XILINX_M2MSC_FMT_RGB8,
-	XILINX_M2MSC_FMT_YUV8,
-	XILINX_M2MSC_FMT_Y_UV10,
-	XILINX_M2MSC_FMT_Y_UV10_420,
-	XILINX_M2MSC_FMT_Y8,
-	XILINX_M2MSC_FMT_Y10,
-	XILINX_M2MSC_FMT_BGRA8,
-	XILINX_M2MSC_FMT_BGRX8,
-	XILINX_M2MSC_FMT_UYVY8,
-	XILINX_M2MSC_FMT_BGR8,
+	XILINX_M2MSC_FMT_RGBX8		= 10,
+	XILINX_M2MSC_FMT_YUVX8		= 11,
+	XILINX_M2MSC_FMT_YUYV8		= 12,
+	XILINX_M2MSC_FMT_RGBX10		= 15,
+	XILINX_M2MSC_FMT_YUVX10		= 16,
+	XILINX_M2MSC_FMT_Y_UV8		= 18,
+	XILINX_M2MSC_FMT_Y_UV8_420	= 19,
+	XILINX_M2MSC_FMT_RGB8		= 20,
+	XILINX_M2MSC_FMT_YUV8		= 21,
+	XILINX_M2MSC_FMT_Y_UV10		= 22,
+	XILINX_M2MSC_FMT_Y_UV10_420	= 23,
+	XILINX_M2MSC_FMT_Y8		= 24,
+	XILINX_M2MSC_FMT_Y10		= 25,
+	XILINX_M2MSC_FMT_BGRX8		= 27,
+	XILINX_M2MSC_FMT_UYVY8		= 28,
+	XILINX_M2MSC_FMT_BGR8		= 29,
 };
 
 /**
@@ -153,18 +150,6 @@ static const struct xm2msc_fmt formats[] = {
 		.xm2msc_fmt = XILINX_M2MSC_FMT_YUYV8,
 		.num_planes = 1,
 	},
-	{
-		.name = "bgr8888",
-		.fourcc = 0,
-		.xm2msc_fmt = XILINX_M2MSC_FMT_RGBA8,
-		.num_planes = 1,
-	},
-	{
-		.name = "avuy8888",
-		.fourcc = 0,
-		.xm2msc_fmt = XILINX_M2MSC_FMT_YUVA8,
-		.num_planes = 1,
-	},
 	{
 		.name = "xbgr2101010",
 		.fourcc = V4L2_PIX_FMT_XBGR30,
@@ -177,24 +162,12 @@ static const struct xm2msc_fmt formats[] = {
 		.xm2msc_fmt = XILINX_M2MSC_FMT_YUVX10,
 		.num_planes = 1,
 	},
-	{
-		.name = "nv16",
-		.fourcc = V4L2_PIX_FMT_NV16M,
-		.xm2msc_fmt = XILINX_M2MSC_FMT_Y_UV8,
-		.num_planes = 2,
-	},
 	{
 		.name = "nv16",
 		.fourcc = V4L2_PIX_FMT_NV16,
 		.xm2msc_fmt = XILINX_M2MSC_FMT_Y_UV8,
 		.num_planes = 2,
 	},
-	{
-		.name = "nv12",
-		.fourcc = V4L2_PIX_FMT_NV12M,
-		.xm2msc_fmt = XILINX_M2MSC_FMT_Y_UV8_420,
-		.num_planes = 2,
-	},
 	{
 		.name = "nv12",
 		.fourcc = V4L2_PIX_FMT_NV12,
@@ -213,24 +186,12 @@ static const struct xm2msc_fmt formats[] = {
 		.xm2msc_fmt = XILINX_M2MSC_FMT_YUV8,
 		.num_planes = 1,
 	},
-	{
-		.name = "xv20",
-		.fourcc = V4L2_PIX_FMT_XV20M,
-		.xm2msc_fmt = XILINX_M2MSC_FMT_Y_UV10,
-		.num_planes = 2,
-	},
 	{
 		.name = "xv20",
 		.fourcc = V4L2_PIX_FMT_XV20,
 		.xm2msc_fmt = XILINX_M2MSC_FMT_Y_UV10,
 		.num_planes = 2,
 	},
-	{
-		.name = "xv15",
-		.fourcc = V4L2_PIX_FMT_XV15M,
-		.xm2msc_fmt = XILINX_M2MSC_FMT_Y_UV10_420,
-		.num_planes = 2,
-	},
 	{
 		.name = "xv15",
 		.fourcc = V4L2_PIX_FMT_XV15,
@@ -249,12 +210,6 @@ static const struct xm2msc_fmt formats[] = {
 		.xm2msc_fmt = XILINX_M2MSC_FMT_Y10,
 		.num_planes = 1,
 	},
-	{
-		.name = "rgb8888",
-		.fourcc = 0,
-		.xm2msc_fmt = XILINX_M2MSC_FMT_BGRA8,
-		.num_planes = 1,
-	},
 	{
 		.name = "xrgb8888",
 		.fourcc = V4L2_PIX_FMT_XBGR32,
@@ -1064,11 +1019,8 @@ xm2msc_cal_stride(unsigned int width, enum xm2msc_pix_fmt xfmt)
 	switch (xfmt) {
 	case XILINX_M2MSC_FMT_RGBX8:
 	case XILINX_M2MSC_FMT_YUVX8:
-	case XILINX_M2MSC_FMT_RGBA8:
-	case XILINX_M2MSC_FMT_YUVA8:
 	case XILINX_M2MSC_FMT_RGBX10:
 	case XILINX_M2MSC_FMT_YUVX10:
-	case XILINX_M2MSC_FMT_BGRA8:
 	case XILINX_M2MSC_FMT_BGRX8:
 		stride = width * 4;
 		break;
-- 
2.31.1

